src/codegen/codegen_info.cpp

```diff
@@ -17,6 +19,48 @@ namespace codegen {
 
 using visitor::VarUsageVisitor;
 
+LongitudinalDiffusionInfo::LongitudinalDiffusionInfo(
+    const std::shared_ptr<ast::Name>& volume_index_name,
+    std::shared_ptr<ast::Expression> volume_expr,
+    const std::shared_ptr<ast::Name>& rate_index_name,
+    std::shared_ptr<ast::Expression> rate_expr)
+    : volume_index_name(volume_index_name ? volume_index_name->get_node_name() : std::string{})
+    , volume_expr(std::move(volume_expr))
+    , rate_index_name(rate_index_name ? rate_index_name->get_node_name() : std::string{})
+    , rate_expr(std::move(rate_expr)) {}
+
+std::shared_ptr<ast::Expression> LongitudinalDiffusionInfo::volume(
+    const std::string& index_name) const {
+    return substitute_index(index_name, volume_index_name, volume_expr);
+}
+std::shared_ptr<ast::Expression> LongitudinalDiffusionInfo::diffusion_rate(
+    const std::string& index_name) const {
+    return substitute_index(index_name, rate_index_name, rate_expr);
+}
+
+double LongitudinalDiffusionInfo::dfcdc(const std::string& /* index_name */) const {
+    // Needed as part of the Jacobian to stabalize
+    // the implicit time-integration. However,
+    // currently, it's set to `0.0` for simplicity.
```

Comment:
This comment can go in the header as it's not very obvious what the function does from the name alone.

Reply:
This one's tricky. I read a lot more code than doxygen documentation; as such I see the implementation more than the declaration. I would prefer to see the explanation right next to the `return 0.0` (which without comment is really strange).

One could completely inline `dfcdc`.

Reply:
Fair enough, though in that case I think it's best to put this above the function signature so it shows up in the docs (I think doxygen then correctly picks it up). Also, possible small typo, should be "stabilize".